### PR TITLE
Add DagModel attributes before dumping DagDetailSchema for get_dag_details API endpoint

### LINKED ISSUE #33482: The `/dags/{dag_id}/details` endpoint returns less data than is documented
### What do you see as an issue?

The [/dags/{dag_id}/details](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/post_set_task_instances_state) endpoint of the REST API does not return all of the keys that are listed in the documentation. If I run `curl -X GET localhost:8080/api/v1/dags/{my_dag}/details`, then compare the results with the results in the documentation, you can see the following missing keys:

```python
>>> for key in docs.keys():
...     if not key in actual.keys():
...             print(key)
...
root_dag_id
last_parsed_time
last_pickled
last_expired
scheduler_lock
timetable_description
has_task_concurrency_limits
has_import_errors
next_dagrun
next_dagrun_data_interval_start
next_dagrun_data_interval_end
next_dagrun_create_after
template_search_path
```

### Solving the problem

Either remove these keys from the documentation or fix the API endpoint

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by RNHTTR**
Closes: #33482

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_connexion/endpoints/dag_endpoint.py [hunk 1]
```diff
@@ -66,6 +66,11 @@ def get_dag_details(*, dag_id: str) -> APIResponse:
     dag: DAG = get_airflow_app().dag_bag.get_dag(dag_id)
     if not dag:
         raise NotFound("DAG not found", detail=f"The DAG with dag_id: {dag_id} was not found")
+    dag_model: DagModel = DagModel.get_dagmodel(dag_id=dag_id)
+    for key, value in dag_model.__dict__.items():
+        if not key.startswith("_") and not hasattr(dag, key):
+            setattr(dag, key, value)
```
Comment (uranusjr) [inline airflow/api_connexion/endpoints/dag_endpoint.py]: Not sure if modifying the DAG object in-place is the right thing to do, I think the DAG bag reuses the objects in different calls. The opposite is likely better, fetching a DagModel object first and set attributes from a DAG.

Comment (RNHTTR) [reply airflow/api_connexion/endpoints/dag_endpoint.py]: Would it make more sense to deepcopy the dag object?

Comment (uranusjr) [reply airflow/api_connexion/endpoints/dag_endpoint.py]: Yes but why not use the DagModel object instead? That would also better match other DAG endpoints, which all use the db model instead.

File: airflow/api_connexion/openapi/v1.yaml
Comment (uranusjr) [inline airflow/api_connexion/openapi/v1.yaml:1]: Are quote changes in this file needed? They are obscuring the actual addition…

File: airflow/api_connexion/schemas/dag_schema.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ class DAGDetailSchema(DAGSchema):
     is_active = fields.Method("get_is_active", dump_only=True)
     is_paused_upon_creation = fields.Boolean()
     end_date = fields.DateTime(dump_only=True)
-    template_search_path = fields.String(dump_only=True)
+    template_searchpath = fields.String(dump_only=True)
```
Comment (uranusjr) [inline airflow/api_connexion/schemas/dag_schema.py:112]: Can we change this without backward compatibility concerns?

**Comment by uranusjr** [comment]
A couple of minor questions, otherwise lgtm.

File: airflow/api_connexion/schemas/dag_schema.py (hunk 2 above)
Comment (RNHTTR) [reply airflow/api_connexion/schemas/dag_schema.py:112]: Yeah -- I'm pretty sure this never worked. In the [DAG model](https://github.com/apache/airflow/blob/main/airflow/models/dag.py#L442), it's `template_searchpath`, so if I understand correctly, this field should now be available where the `dag_schema` is referenced.

File: airflow/api_connexion/openapi/v1.yaml
Comment (RNHTTR) [reply airflow/api_connexion/openapi/v1.yaml:1]: I'm not sure what you mean by quote changes

**Review by uranusjr — APPROVED** [review]

File: airflow/api_connexion/openapi/v1.yaml
Comment (pierrejeambrun) [reply airflow/api_connexion/openapi/v1.yaml:1]: There are an important number of lines changed located in the OpenAPI spec. All simple quotes are moved to double quotes.

I also agree that it makes the addition harder to review, I suggest we do this in a separate PR (I don't think it is needed in this particular PR).

Makes me realize that we do not have yamllint activated on our OpenAPI Spec to enforce common style.

**Review by eladkal — APPROVED** [review]

**Pull request merged**